purchases/src/test/java/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -1781,6 +1783,264 @@ class PurchasesTest {
         }
     }
 
+    @Test
+    fun `login with the same appUserID as the current, fetches purchaserInfo and calls onSuccess if successful`() {
+        setup()
+        val appUserID = "myUser"
+        every { mockCache.isPurchaserInfoCacheStale(appUserID, any()) } returns true
+        every { mockCache.setPurchaserInfoCacheTimestampToNow(appUserID) } just Runs
+        every { mockIdentityManager.currentAppUserID } returns appUserID
+
+        val mockCompletion = mockk<LogInCallback>(relaxed = true)
+
+        purchases.logIn(appUserID, mockCompletion)
+
+        verify {
```

Comment:
I like to specify the number of calls that should be verified. It's possible that `mockBackend.getPurchaserInfo` is called from another part of the code, and `logIn` never calls it, and this test will still pass. I think making sure that what is being passed to the `mockCompletion` is what's fetched from the backend could also work.

Reply:
agreed that explicit > implicit. I was just following the style that was already in the file, but I'll update for all the stuff in this PR